visual_reasoning_nemo_collection/image/resnet.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) IBM Corporation 2018
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import torch.nn as nn
+import torchvision as tv
+from nemo.backends.pytorch.nm import TrainableNM
+from nemo.core.neural_types import ChannelType, NeuralType
+from nemo.utils.decorators import add_port_docs
+from nemo.core import DeviceType
+
+class Resnet50(TrainableNM):
+    """
+    Wraps Resnet50 from torchvision.
+    Parametrizable last fc layer dim.
+    """
+
+    @property
+    @add_port_docs()
+    def input_ports(self):
+        """Returns definitions of module input ports.
+        """
+        return {
+            "x": NeuralType(('B', 'C', 'H', 'W'), ChannelType()),
+        }
+
+    @property
+    @add_port_docs()
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        """
+        return {
+            "output": NeuralType(('B', 'D'), ChannelType()),
+        }
+
+    def __init__(self, output_size=1000):
```

Comment:
1000? Why 1000, buddy? ;) We like round numbers, like 512, 1024 etc. ;)

Reply:
Because that's the number of outputs that the ResNets from torchvision have, because ImageNet. Initially the output size is the number of classes.

Reply:
Yeah, but we are not talking about the last (logit) layer size, but the embeddings size? Or maybe I misunderstood the code... and hmmm, it seems I cannot see it anymore ;)

Reply:
I rebased, you reopen the file from the new commit reference

Reply:
https://github.com/pytorch/vision/blob/e6b4078ec73c2cf4fd4432e19c782db58719fb99/torchvision/models/resnet.py#L124-L128